modules/ppcp-button/src/Assets/SmartButton.php

```diff
@@ -331,6 +331,7 @@ private function render_button_wrapper_registrar(): bool {
 		if (
 			is_cart()
 			&& ! $not_enabled_on_cart
+			&& ! $this->is_cart_price_total_zero()
```

Comment:
Maybe better just check once at the start of function and `return`?

Reply:
That will not be better I think because there is also another [action for single_product_renderer_hook](https://github.com/woocommerce/woocommerce-paypal-payments/blob/164b74fdcba08b31c6543b224752bf9fde9eb7fc/modules/ppcp-button/src/Assets/SmartButton.php#L353) and for this one, another, [is_product_price_total_zero](https://github.com/woocommerce/woocommerce-paypal-payments/blob/164b74fdcba08b31c6543b224752bf9fde9eb7fc/modules/ppcp-button/src/Assets/SmartButton.php#L351) function is used to check

Reply:
anyway I have improved this a little here: 6de1aa2ae8f24966b4becea855ccb172d0c0ad52
I have just put action for `single_product_renderer_hook` at the top of the function and then used early return for the rest of the function